state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -1147,6 +1147,31 @@ impl<T: Config> PoolApi for Pallet<T> {
 		Pools::<T>::iter_keys().map(|asset_pair| asset_pair.assets()).collect()
 	}
 
+	fn cancel_all_limit_orders(account: &Self::AccountId) -> DispatchResult {
+		// Collect to avoid undefined behaviour (See StorageMap::iter_keys documentation)
+		for asset_pair in Pools::<T>::iter_keys().collect::<Vec<_>>() {
+			let pool = Pools::<T>::get(asset_pair).unwrap();
```

Comment:
```suggestion
for (asset_pair, pool) in Pools::<T>::iter().collect::<Vec<_>>() {
```

Reply:
I see you copied that code... guess we can change it at the source too.

Reply:
The comment is about collecting, which this still does so avoids what the iter_keys() doc is talking about, but having all pools in memory is a fair reason not to - we could comment that here.

Reply:
> having all pools in memory is a fair reason not to

This seems to be like a case of premature optimisation though? It is not clear to me that having a tiny memory spike (we are likely talking about well under 1MB here, right?) is worse than the overhead from looking up entries individually. When it is not clear, IMO we should prefer the cleaner solution (e.g. one that doesn't require extra comments) like what Kyle suggested.

I will add a comment and leave the code as is for now (so we can merge it now), but if we don't mind either way, I might change this at some point later.